great_expectations/core/metric_function_types.py

```diff
@@ -1,27 +1,76 @@
 import enum
 import logging
 
+from great_expectations.core._docs_decorators import public_api
+
 logger = logging.getLogger(__name__)
 
 
+@public_api
 class MetricFunctionTypes(enum.Enum):
+    """Enum type, whose members depict the nature of return value of a metric implementation function (defined for
+    a specified "ExecutionEngine" subclass) that is the final result (rather than a Callable for deferred execution).
+
+    The available types are:
+    - VALUE -- metric implementation function returns a value computed over a dataset represented by "Domain" (e.g., a
+      statistic on row values of a column).
+    - MAP_VALUES (unused in current version) -- metric implementation function returns a mapping between every "Domain"
+      value and the result of a transformation of the corresponding "Domain" value.
+    - WINDOW_VALUES (unused in current version) -- metric implementation function returns the result of applying a
+      specified windowing operation over "Domain" values.
+    - AGGREGATE_VALUE (unused in current version) -- metric implementation function returns the result of applying a
```

Comment:
For these unused values, would it be possible to use the `deprecated_argument` decorator?

```
@deprecated_argument(argument_name="batches", version="0.14.0")
```